src/Domain/ReadySession/Model/ReadySessionId.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+declare(strict_types=1);
+
+namespace StepupReadId\Domain\ReadySession\Model;
+
+use Ramsey\Uuid\Uuid;
+use Ramsey\Uuid\UuidInterface;
```

Comment:
Strictly speaking, the Ramsey UUID (infra) utility should not be used in the Domain layer. This way we couple our domain layer to a concrete library. Or do you see this differently?

Reply:
Yes, but SOLID speaks about modules should not depend on modules in low levels. In this case it is another VO (same level of abstraction), same thing if you need an Identity VO from another Bounded Context. In this case wins the `Code reuse` principle (IMHO). Ramsey is a stable and extended used library.

Reply:
That's another interesting way of looking at it. Personally I prefer not using these in the domain layer. But I have no objections to leaving it like this. If I'd be faced with your problem I'd maybe encapsulate the UUID logic in a domain service or something like that, or maybe the application layer should be tasked with this?

Reply:
In this case, UUID is just a Value Object. Think about a use case that you need to use the Money VO. Why do you are going to implement your own logic or use a service when already have a Money library implementation? A Domain Service I think should be only used to implement business rules  and not as a Factory.